torch/csrc/jit/export.cpp

```diff
@@ -580,23 +582,47 @@ void ScriptModuleSerializer::serialize(
 }
 
 void ScriptModuleSerializer::writeLibs(torch::ModelDef* model_def) {
-  auto lib_def = model_def->mutable_libs();
-  std::ostringstream lib_stream;
-  lib_stream << "op_version_set = " << op_version_set << "\n";
-  // Convert all the classes that
+  // Convert all the classes that this model depends on
   for (const auto& class_type : class_table_) {
     convertClass(class_type, model_def);
   }
 
-  for (const auto& c : converted_classes_) {
-    lib_stream << *c << "\n";
+  // Mapping of filename => src. We need this because multiple clases may go in
+  // the same file (e.g. foo.bar.Baz and foo.bar.Qux)
+
+  // Aggregate classes into files by their qualified names
+  std::unordered_map<std::string, std::string> fileToSrc;
+  for (const auto& item : converted_classes_) {
+    const auto& class_type = item.key();
+    const auto& class_src = item.value();
+
+    // For the type, foo.bar.Baz
+    const std::string filename =
+        ImportExportHelpers::qualifierToPath(class_type->qualifier());
+    // End state: filename is "foo/bar.py", in which we will define a class
+    // named Baz
+    fileToSrc[filename] += class_src;
   }
 
-  torch::RecordRef* lib_record = lib_def->mutable_torchscript_arena();
-  const auto filename = "libs.py";
-  const auto& lib_str = lib_stream.str();
-  writer_.writeRecord(filename, lib_str.c_str(), lib_str.size());
-  lib_record->set_key(filename);
+  // Write out the files. We still have to do this in converted_classes_ order,
+  // to maintain dependency order.
+  for (const auto& item : converted_classes_) {
+    const ClassTypePtr& class_type = item.key();
+    const std::string filename =
+        ImportExportHelpers::qualifierToPath(class_type->qualifier());
+    const std::string& src = fileToSrc.at(filename);
+
+    std::ostringstream lib_stream;
+    lib_stream << "op_version_set = " << op_version_set << "\n";
```

Comment:
aside: we should move the op_version_set into the json file, now that there are so many source files. This would also clear up a bunch of tests where we have to manually prepend this thing.